source/shared/cpp/ObjectModel/Container.cpp

```diff
@@ -108,6 +124,13 @@ std::shared_ptr<Container> Container::Deserialize(const Json::Value& value)
     // Parse Items
     auto cardElements = ParseUtil::GetElementCollection<BaseCardElement>(value, AdaptiveCardSchemaKey::Items, Container::CardElementParsers, true);
     container->m_items = std::move(cardElements);
+
+    Json::Value selectActionValue = ParseUtil::ExtractJsonValue(value, AdaptiveCardSchemaKey::SelectAction, false);
+    if (!selectActionValue.empty())
+    {
+        container->SetSelectAction(ParseUtil::GetActionFromJsonValue<BaseActionElement>(selectActionValue, BaseCardElement::ActionParsers));
+    }
```

Comment:
Consider refactoring this code.  Possibly create helper methods in BaseCardElement to parse Card Element and Action.  This would allow you to keep ActionParsers private and remove all the duplicated Card Element Parsers.